remoting/src/main/java/org/apache/rocketmq/remoting/util/ServiceProvider.java

```diff
@@ -131,6 +132,53 @@ protected static InputStream getResourceAsStream(ClassLoader loader, String name
         return services;
     }
 
+    public static Map<String, String> loadPath(String path) {
+        LOG.info("Load path looking for a resource file of name [{}] ...", path);
+        Map<String, String> pathMap = new HashMap<String, String>();
+        try {
+            final InputStream is = getResourceAsStream(getContextClassLoader(), path);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *is*. The resource is referenced in statements at the following lines: 140, 143, 145, 147, 159. The resource closure statement is at line: 161. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by BufferedReader.readLine. Either a) close *is* in a try-finally block or b) close the resource by declaring *is* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).